metadata-ingestion/src/datahub/utilities/source_helpers.py

```diff
@@ -103,10 +104,10 @@ def auto_stale_entity_removal(
     yield from stale_entity_removal_handler.gen_removed_entity_workunits()
 
 
-def auto_workunit_reporter(
-    report: SourceReport,
-    stream: Iterable[MetadataWorkUnit],
-) -> Iterable[MetadataWorkUnit]:
+T = TypeVar("T", bound=WorkUnit)
+
+
+def auto_workunit_reporter(report: SourceReport, stream: Iterable[T]) -> Iterable[T]:
```

Comment:
I would love to get to a point where sources can also generate MCPs and MCEs directly, and the helper wraps them into WorkUnits as necessary